plugins/woocommerce-admin/client/wp-admin-scripts/shipping-settings-region-picker/currency-context.js

```diff
@@ -5,14 +5,39 @@ import { useContext, useEffect } from '@wordpress/element';
 import { CurrencyContext } from '@woocommerce/currency';
 import { numberFormat } from '@woocommerce/number';
 
+/**
+ * Format all numbers with respect to shipping formula.
+ */
+export const numberFormatWithShippingFormula = ( config, number ) => {
+	if ( typeof number === 'number' ) {
+		return numberFormat( config, number );
+	}
+
+	if ( typeof number === 'string' ) {
+		/**
+		 * \b: Ensures that we're capturing whole numbers (word boundaries).
+		 * \d+: Matches one or more digits.
+		 * (\.\d+)?: Optionally matches a decimal point followed by one or more digits.
+		 * (?![^[]*\]): A negative lookahead to avoid numbers inside square brackets.
+		 * g: Global flag to match all instances in the string.
+		 */
+		return number.replace( /(\b\d+(\.\d+)?\b)(?![^[]*\])/g, ( n ) =>
+			numberFormat( config, n )
```

Comment:
I'm not entirely sure we should format numbers in the formula.

`50000000.00 + (([qty]*2.00+1.00)(5.00*10.00))(1.00)` becomes the following. Its probably rare but I'm not sure how much we gain here. 

```
50,000,000.00 + (([qty]*2.00+1.00)(5.00*10.00))(1.00)
```

Reply:
Then again, it does seem to work with the formatted values. I'm still of the mind that this introduces complexity and we can remove it. What do you think?

<img width="502" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1922453/5aad43d9-485a-4ad5-b2cd-a4440f854e79">

Reply:
Good point, I tend to agree now. I'll simplify to only handle formatting when it's exclusively a number. Thanks for the feedback!